File: c/1433.c
bool checkIfCanBreak(char* s1, char* s2) {
    int f[26] = {0};
    for (int i = strlen(s1) - 1; i >= 0; i--) {
        f[s1[i] - 97]++;
        f[s2[i] - 97]--;
    }
    int p1 = 0, p2 = 0;
    bool f1 = true, f2 = true;
    for (int i = 0; i < 26; i++) {
        p1 += f[i];
        p2 -= f[i];
        if (p1 < 0)
            f1 = false;
        if (p2 < 0)
            f2 = false;
    }
    return f1 || f2;
}

File: java/1433.java
class Solution {
    public boolean checkIfCanBreak(String s1, String s2) {
        int f[] = new int[26];
        for (int i = s1.length() - 1; i >= 0; i--) {
            f[s1.charAt(i) - 'a']++;
            f[s2.charAt(i) - 'a']--;
        }
        return check1(f) || check2(f);
    }

    private boolean check1(int f[]) {
        int p = 0;
        for (int i = 0; i < 26; i++) {
            p += f[i];
            if (p < 0)
                return false;
        }
        return true;
    }

    private boolean check2(int f[]) {
        int p = 0;
        for (int i = 0; i < 26; i++) {
            p -= f[i];
            if (p < 0)
                return false;
        }
        return true;
    }
}
